CMakeLists.txt

```
@@ -943,6 +943,19 @@ if(CURL_USE_RUSTLS)
     string(APPEND CMAKE_C_FLAGS " ${RUSTLS_CFLAGS}")
   endif()
 
+  if(RUSTLS_VERSION AND RUSTLS_VERSION VERSION_GREATER_EQUAL "0.15")
+    set(RUSTLS_HAS_HPKE TRUE)
+  else()
+    cmake_push_check_state()
+    list(APPEND CMAKE_REQUIRED_INCLUDES   "${RUSTLS_INCLUDE_DIRS}")
+    list(APPEND CMAKE_REQUIRED_LIBRARIES  "${RUSTLS_LIBRARIES}")
```

```suggestion
    list(APPEND CMAKE_REQUIRED_LIBRARIES  "${RUSTLS_LIBRARIES}")
    curl_required_libpaths("${RUSTLS_LIBRARY_DIRS}")
```
This might solve it for `pkg-config`.

---

That did the trick, thank you! :partying_face:  I used that fix to implement a unified check in `CMakeLists.txt` and removed the `RUSTLS_VERSION` check.

Would you prefer I keep that to use the "optimize out this slow check if the version if known:" strategy you suggested? LMK.

---

I think it'd be nice, to give it a speed-bump with `pkg-config` builds. (tests might take up to several seconds each on Windows.)
Maybe best to add it right before the block above (instead of `FindRustls.cmake`)?

Also maybe move the detection block inside an `if(NOT DEFINED HAVE_RUSTLS_SUPPORTED_HPKE)`? As done at similar places. (mostly a cosmetic in this case)

---

> Maybe best to add it right before the block above (instead of FindRustls.cmake)?

Done :ballot_box_with_check: 

> Also maybe move the detection block inside an if(NOT DEFINED HAVE_RUSTLS_SUPPORTED_HPKE)? As done at similar places. (mostly a cosmetic in this case)

Done :ballot_box_with_check: 

I redid the CMake tests & updated the PR description after verifying that none of the pkg-config based builds that have `RUSTLS_VERSION` defined did the slow `check_symbol_exists()` check.